tiffslide/tiffslide.py

```diff
@@ -375,32 +377,77 @@ def read_region(
             size (width, height) of the requested region
         as_array :
             if True, return the region as numpy array
+        padding :
+            if True, will ensure that the size of the returned image is deterministic.
         """
         base_x, base_y = location
         base_w, base_h = self.dimensions
-        level_w, level_h = self.level_dimensions[level]
```

Comment:
We should make clear that the most common code path is the one where no padding is required at all.

picking the wrong level is probably the least common code path even.

I would start all of this with something like:

```python
        base_x, base_y = location
        base_w, base_h = self.dimensions
        _rw, _rh = size
        axes = self.properties["tiffslide.series-axes"]

        try:
            level_w, level_h = self.level_dimensions[level]
        except IndexError:
            if not padding:
                raise
            warn(
                f"level={level} is out-of-bounds, but padding is requested",
                stacklevel=2,
            )

            zarray: zarr.core.Array
            if isinstance(self.ts_zarr_grp, zarr.core.Array):
                zarray = self.ts_zarr_grp
            else:
                zarray = self.ts_zarr_grp["0"]

            if axes == "YXS":
                depth = zarray.shape[2]
            elif axes == "CYX":
                depth = zarray.shape[0]
            else:
                raise NotImplementedError
            return np.zeros((_rh, _rw, depth), dtype=zarray.dtype)

```

You'll see that you will be able to keep most of the existing code further below intact, too, if you do it this way.